src/logging.ts

```diff
@@ -0,0 +1,102 @@
+import { Account } from "./types";
+import { BigQuery } from "@google-cloud/bigquery";
+import { config } from "./config";
+
+///Depending on your setup, you can use either Caspian or BigQuery for logging. Other databases may be added in the future
+export async function insertIntoCaspian(
+  account: Account,
+  amount: number,
+  reqBody: any,
+  networkId: number
+): Promise<any> {
+  const dataPayload = [
+    {
+      user_agent: reqBody.userAgent || "",
+      usage_context: reqBody.usageContext || "",
+      memos: reqBody.memos || [],
+      account: account.xAddress,
+      amount: amount,
+      network: networkId,
+    },
+  ];
+
+  const postData = JSON.stringify({
+    producerName: config.CASPIAN_PRODUCER_NAME,
+    entityName: config.CASPIAN_ENTITY_NAME,
+    schemaType: config.CASPIAN_SCHEMA_TYPE,
+    schemaVersion: Math.round(config.CASPIAN_SCHEMA_VERSION),
+    data: dataPayload,
+    timestamp: Date.now(),
+  });
+
+  try {
+    const response = await fetch(config.CASPIAN_ENDPOINT, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "x-api-key": config.CASPIAN_API_KEY,
+      },
+      body: postData,
+    });
+
+    if (response.ok) {
+      return await response.json();
+    } else {
+      const errorMessage = await response.text();
+      throw new Error(`Failed to send data to Caspian: ${errorMessage}`);
```

Comment:
```suggestion
      throw new Error(`${errorMessage}`);
```
As written, this will include that message regardless since it will throw an error immediately with your try / catch.